test/pythonml/test_workflow_integration.py

```diff
@@ -0,0 +1,175 @@
+        to_run = []
+        for to_copy in in_test:
+            source = asset_path + unit_shortnames[to_copy]
+            destination = unit_shortnames[to_copy]
+            to_run.append(destination)
+            shutil.copy(source, destination)
+
+        for file in to_run:
+            pipes = subprocess.Popen((sys.executable, file), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            stdout, stderr = pipes.communicate()
+
+            sys.stdout.write(stdout.decode())
+
+            self.assertFalse(stderr, f"\nSTDERR:\n{stderr.decode()}")
+
+    def set_to_training_phase(self):
+        with open("settings.py", "r") as inp:
+            lines = inp.readlines()
+            sub_partial = functools.partial(re.sub, "(?<=is_workflow_running_to_predict\s=\s)False", "True")
+            edited_lines = "".join(map(sub_partial, lines))
+
+        with open("settings.py", "w") as outp:
+            for line in edited_lines:
+                outp.write(line)
+
+    def run_tests(self, units_in_test):
+
+        # Training Phase
+        self.simulate_workflow(units_in_test)
+        if self.plot_unit in units_in_test:
+            self.assertTrue(os.path.exists(self.plot_name))
+
+        # Reconfigure for predictions
+        self.set_to_training_phase()
+
+        # Predict Phase
+        self.simulate_workflow(units_in_test)
+        self.assertTrue(os.path.exists("predictions.csv"))
+
+    def tearDown(self) -> None:
+        for file in os.listdir():
+            if (file in files_to_remove) or any([file.endswith(ext) for ext in extensions_to_remove]):
+                try:
+                    os.remove(file)
+                except (IsADirectoryError, PermissionError):
+                    shutil.rmtree(file)
+                except FileNotFoundError:
+                    pass
+
+
+class TestRegression(BasePythonMLTest):
+    category = "regression"
+    plot_name = "my_parity_plot.png"
+    plot_unit = "POS_plotParity"
+
+    @parameterized.expand(tests_regression, testcase_func_name=custom_name_func(names_regression),
+                          skip_on_empty=True)
+    def test_workflows(self, *units_in_test):
+        self.run_tests(units_in_test)
+
+
+class TestClassification(BasePythonMLTest):
+    category = "classification"
+    plot_name = "my_roc_curve.png"
+    plot_unit = "POS_plotROC"
+
+    @parameterized.expand(tests_classification, testcase_func_name=custom_name_func(names_classification),
+                          skip_on_empty=True)
+    def test_workflows(self, *units_in_test):
+        self.run_tests(units_in_test)
+
+
+class TestClustering(BasePythonMLTest):
+    category = "clustering"
+    plot_name = "my_clusters.png"
+    plot_unit = "POS_plotClust"
+
+    @parameterized.expand(tests_clustering, testcase_func_name=custom_name_func(names_clustering),
+                          skip_on_empty=True)
+    def test_workflows(self, *units_in_test):
+        self.run_tests(units_in_test)
```

Comment:
This is all a significant speed bump for a reader to understand, given the lack of explanation.  Please comment at least one of these classes, to explain what's going on, especially what `@parameterized` is doing with custom_name_func

Because of the meta-programming nature of this code, we need more commentary explaining what's happening, not less.  It's important to be able to understand how the tests are run.

Reply:
Added a ton of documentation to the file